src/server/journal/executor.cc

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "server/journal/executor.h"
+
+#include "base/logging.h"
+#include "server/main_service.h"
+
+namespace dfly {
+
+JournalExecutor::JournalExecutor(Service* service) : service_{service} {
+}
+
+void JournalExecutor::Execute(journal::ParsedEntry&& entry) {
+  if (entry.payload) {
+    io::NullSink null_sink;
+    ConnectionContext conn_context{&null_sink, nullptr};
+    conn_context.is_replicating = true;
+    conn_context.journal_emulated = true;
+    conn_context.conn_state.db_index = entry.dbid;
+
+    auto span = CmdArgList{entry.payload->data(), entry.payload->size()};
+
+    VLOG(0) << "EXECUTING";
+    for (auto s : span)
+      VLOG(0) << std::string_view{s.data(), s.size()};
+    VLOG(0) << "====";
+
+    service_->DispatchCommand(span, &conn_context);
```

Comment:
We need to somehow get the error back from the command.

We do now want it to write reponses for optimal performance. I guess we can either:
- Replace it with a custom sink that looks for the RESP error sequence and reports it
- Add an overloaded serializer (they are virtual, so it should be possible) that reports only errors.

The second approach seems to be better